fastfilter/src/main/java/org/fastfilter/xor/XorSimple.java

```diff
@@ -51,7 +51,7 @@ boolean map(long[] keys, long seed, long[] stack) {
             }
         }
         int si = 0;
-        while (si < 2 * keys.length) {
+        while (si < 2 * keys.length && qi > 0) {
```

Comment:
The XorSimple is (on purpose) the most simple possible implementation. It is only for educational purposes. It shouldn't have any performance improvements. But of course it shouldn't have any bugs either... So not sure if this is really needed?

Reply:
This fixes a bug where qi goes negative in the loop, rather than fail to map. This occurred for the first time with the seed in the regression tests, but cannot be reproduced because of the unseeded calls to new Random().nextLong() (which #13 aims to fix). Basically there do exist cases where this loop will access `Q[-1]` rather than fail to map and try again, but it's nondeterministic.

Reply:
I suggest you omit this change from your PR and, instead, report it as a bug. It seems likely to me that your fix (qi > 0) is just patching over a logical error of some kind. It would be better to identify the logical error.

Reply:
OK I will do that.